iron-list.html

```diff
@@ -1390,11 +1390,14 @@
 
       forceUpdate = forceUpdate || this._scrollHeight === 0;
       forceUpdate = forceUpdate || this._scrollPosition >= this._estScrollHeight - this._physicalSize;
+      // Probably a bug that style.height is a string like '100px' and it is being compared to
+      // a number?
       forceUpdate = forceUpdate || this.grid && this.$.items.style.height < this._estScrollHeight;
 
       // Amortize height adjustment, so it won't trigger large repaints too often.
       if (forceUpdate || Math.abs(this._estScrollHeight - this._scrollHeight) >= this._optPhysicalSize) {
         this.$.items.style.height = this._estScrollHeight + 'px';
+        this.updateViewportBoundaries();
```

Comment:
In my example there is no CSS rule to specify the size, on purpose. I would understand your point better, if resizing the window had absolutely no effect when I don't have an explicit size in CSS (keep showing 3 elements only).

The fact that resizing the window results in more elements shown, even though no size is specified in CSS makes it very confusing. Which brings me to my question, is iron-list using obsolete cached values on purpose? Is this CL wrong from a logic perspective, or is it just making things more expensive?